app/moves/middleware.js

```diff
@@ -81,6 +96,24 @@ module.exports = {
 
     next()
   },
+  setMoveTypeNavigation: async (req, res, next) => {
+    const { period, dateRange } = res.locals
+    const { locationId = '' } = req.params
+    try {
+      res.locals.moveTypeNavigation = await Promise.all(
+        cloneDeep(moveTypeNavigationConfig).map(moveType => {
+          return moveService
+            .getMovesCount({ dateRange, status: moveType.filter, locationId })
+            .then(count => {
```

Comment:
This could be a set of chained `then` which could clean up the number of args you need to send to the presenter:

```javascript
cloneDeep(moveTypeNavigationConfig).map(moveType => {
  return moveService
    .getMovesCount({ dateRange, status: moveType.filter, locationId })
    .then(count => {
      return {
        ...moveType,
        count,
        active: moveType.filter === req.params.status,
        url: ...,
      }
    })
    .then(type => moveTypeForFilter(type))
})
```

Reply:
Nice one!!